x-pack/plugins/security/public/management/role_combo_box/role_combo_box.tsx

```diff
@@ -19,43 +36,132 @@ interface Props {
   isDisabled?: boolean;
 }
 
+type Option = EuiComboBoxOptionOption<{
+  isReserved: boolean;
+  isDeprecated: boolean;
+  isSystem: boolean;
+  isAdmin: boolean;
+  deprecatedReason?: string;
+}>;
+
 export const RoleComboBox = (props: Props) => {
   const onRolesChange = (selectedItems: Array<{ label: string }>) => {
     props.onChange(selectedItems.map((item) => item.label));
   };
 
-  const roleNameToOption = (roleName: string) => {
+  const roleNameToOption = (roleName: string): Option => {
     const roleDefinition = props.availableRoles.find((role) => role.name === roleName);
+    const isReserved: boolean = (roleDefinition && isRoleReserved(roleDefinition)) ?? false;
     const isDeprecated: boolean = (roleDefinition && isRoleDeprecated(roleDefinition)) ?? false;
+    const isSystem: boolean = (roleDefinition && isRoleSystem(roleDefinition)) ?? false;
+    const isAdmin: boolean = (roleDefinition && isRoleAdmin(roleDefinition)) ?? false;
     return {
-      color: isDeprecated ? 'warning' : 'default',
+      color: isDeprecated ? 'warning' : isReserved ? 'primary' : undefined,
       'data-test-subj': `roleOption-${roleName}`,
       label: roleName,
       value: {
+        isReserved,
         isDeprecated,
+        isSystem,
+        isAdmin,
+        deprecatedReason: roleDefinition?.metadata?._deprecated_reason,
       },
     };
   };
 
   const options = props.availableRoles.map((role) => roleNameToOption(role.name));
-
   const selectedOptions = props.selectedRoleNames.map(roleNameToOption);
+  const groupedOptions = options.reduce<Record<string, typeof options>>((acc, option) => {
+    const type = option.value?.isDeprecated
+      ? 'deprecated'
+      : option.value?.isSystem
+      ? 'system'
+      : option.value?.isAdmin
+      ? 'admin'
+      : option.value?.isReserved
+      ? 'user'
+      : 'custom';
+    if (!acc[type]) {
+      acc[type] = [];
+    }
+    acc[type].push(option);
+    return acc;
+  }, {});
 
   return (
     <EuiComboBox
```

Comment:
I don't think you made this any worse than it already was, but I think we need an `aria-labelledby` prop here that we can associate the outer `EuiFormRow` with this component. Found this via the `axe` browser extension